github-runner-manager/src/github_runner_manager/http_server.py

```diff
@@ -0,0 +1,82 @@
+#  Copyright 2025 Canonical Ltd.
+#  See LICENSE file for licensing details.
+
+"""The HTTP server for github-runner-manager.
+
+The HTTP server for request to the github-runner-manager.
+"""
+
+from threading import Lock
+
+from flask import Flask, abort
+
+from github_runner_manager.cli_config import Configuration
+
+app = Flask(__name__)
+
+
+# The path under /lock are for tests.
```

Comment:
I would speak in general terms: for "debugging". Also, can we document this somewhere (including the rationale) to avoid having implicit knowledge hanging around?